hazelcast/src/main/java/com/hazelcast/cache/impl/JCacheDetector.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache.impl;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.nio.ClassLoaderUtil;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Utility class to detect existence of JCache 1.0.0 in the classpath.
+ * Earlier versions (1.0.0-PFD, 0.9, 0.5) do not provide the complete JCache API. If a stale version
+ * is detected, a warning will be logged and JCache is considered missing from the classpath.
+ * <p>Related issues:
+ * <ul>
+ * <li>https://github.com/hazelcast/hazelcast/issues/7810</li>
+ * <li>https://github.com/hazelcast/hazelcast/issues/7854</li>
+ * </ul>
+ * </p>
+ */
+public final class JCacheDetector {
+
+    private static final ILogger LOGGER = Logger.getLogger(JCacheDetector.class);
+
+    private static final String JCACHE_CACHING_CLASSNAME =  "javax.cache.Caching";
+    private static final String[] JCACHE_ADDITIONAL_REQUIRED_CLASSES = new String[]{
+            "javax.cache.integration.CacheLoaderException",
+            "javax.cache.integration.CacheWriterException",
+            "javax.cache.processor.EntryProcessorException",
+            "javax.cache.configuration.CompleteConfiguration",
+            };
+
+    private static final AtomicBoolean HINT_LOGGED = new AtomicBoolean();
+
+    // do not allow construction of instances
+    private JCacheDetector() {
+
+    }
+
+    /**
+     * @return {@code true} if JCache 1.0.0 is available in the classpath, otherwise {@code false}
+     */
+    public static boolean isJcacheAvailable(ClassLoader classLoader) {
+        if (!ClassLoaderUtil.isClassAvailable(classLoader, JCACHE_CACHING_CLASSNAME)) {
+            // no cache-api jar in the classpath
+            return false;
+        }
+        for (String className : JCACHE_ADDITIONAL_REQUIRED_CLASSES) {
+            if (!ClassLoaderUtil.isClassAvailable(classLoader, className)) {
+                if (HINT_LOGGED.compareAndSet(false, true)) {
```

Comment:
Do we really need this logging logic? Because the logging check is global but availability check is classloader based. 

I mean, assume that there are at least two different classloader and none of them knows jcache (can't detect jcache classes). In this case, this log message is displayed for only one of them, not for both of them. 

Therefore, I think, logging should be responsibility of the caller of `JCacheDetector`.